apps/aechannel/src/aesc_state_cache.erl

```diff
@@ -96,50 +196,89 @@ start_link() ->
 init([]) ->
     {ok, #st{}}.
 
-handle_call({new, ChId, PubKey, Pid, State}, _From, St) ->
-    case ets:insert_new(?TAB, #ch{id = key(ChId, PubKey), state = State}) of
-        true ->
-            St1 = monitor_fsm(Pid, ChId, PubKey, St),
-            {reply, ok, St1};
-        false ->
-            {reply, {error, exists}, St}
-    end;
-handle_call({reestablish, ChId, PubKey, Pid}, _From, St) ->
-    case try_reestablish_cached(ChId, PubKey) of
+generate_session_key(Password, Salt) ->
```

Comment:
The encryption algo and parameters should be documented as part of the module docs at the top.